pkg/lib/oauth/oidc/handler/handler_end_session.go

```diff
@@ -16,13 +16,41 @@ type WebAppURLsProvider interface {
 	SettingsURL() *url.URL
 }
 
+type IDTokenIssuer interface {
+	ValidateIDToken(idToken string) error
+}
+
+type LogoutSessionManager interface {
+	Logout(session.Session, http.ResponseWriter) error
+}
+
 type EndSessionHandler struct {
-	Config    *config.OAuthConfig
-	Endpoints oidc.EndpointsProvider
-	URLs      WebAppURLsProvider
+	Config         *config.OAuthConfig
+	Endpoints      oidc.EndpointsProvider
+	URLs           WebAppURLsProvider
+	IDTokens       IDTokenIssuer
+	SessionManager LogoutSessionManager
 }
 
 func (h *EndSessionHandler) Handle(s session.Session, req protocol.EndSessionRequest, r *http.Request, rw http.ResponseWriter) error {
+	idTokenHint := req.IDTokenHint()
+	if s != nil && idTokenHint != "" {
+		err := h.IDTokens.ValidateIDToken(idTokenHint)
```

Comment:
Seems we only validate the id token here, but we have not compared it with the current session?
If this is the case, there will be a problem that people can use another peoples'  id token to logout current user.

Reply:
This is quite tricky. Can we have a chat on this.